m2m-transfers/V3/REPWRITERSPECS/BANNO.M2MTRANSFERS.RPT.BATCH

```diff
@@ -0,0 +1,751 @@
+** variable below. I.E: TESTACCOUNTS="0000012345,0000334567,0000004256"
+*]
+ TESTACCOUNTS=""
+
+ CALL INITIALIZATIONS
+ CALL PROGRAMPROMPTS
+ CALL FETCHLETTERFILEPARAMS
+END [SETUP]
+
+SELECT
+[* If M2M cross-account access is turned on then look through all accounts
+** where the account activity date is on or after the reporting start date
+** otherwise, narrow the focus down to only those accounts with an HB user
+** who has last logged into home banking on or after the reporting start date.
+*]
+ (
+  (
+   TESTACCOUNTS="" OR
+   (
+    TESTACCOUNTS<>"" AND
+    CHARACTERSEARCH(TESTACCOUNTS,ACCOUNT:NUMBER)>0
+   )
+  ) AND
+  (
+   ACCOUNT:CLOSEDATE=DATENULL OR
+   ACCOUNT:CLOSEDATE>TARGETSTARTDATE) AND
+   (
+    (
+     XACCTACCESS=FALSE AND
+     ANY PREFERENCE WITH (PREFERENCE:HBUSERNAME<>"" AND
+                          PREFERENCE:HBLASTLOGONDATE>=TARGETSTARTDATE)
+    ) OR 
+    (
+     XACCTACCESS=TRUE AND
+     ACCOUNT:ACTIVITYDATE>=TARGETSTARTDATE
+    )
+   )
+ )
+END [SELECT]
+
+PRINT TITLE="Banno M2M Transfers Report"
+ CALL MAINREPORTHEADER
+ CALL GETSHARETRANSFERRECS
+ CALL GETTRANSACTIONS
+END [PRINT]
+
+PROCEDURE PROGRAMPROMPTS
+[* Prompts used by this program. The default for each prompt is shown
+** in brackets.
+*]
+ TARGETSYMXUSER=CODEREAD("",
+                         "Enter the target SymXchange user number",
+                         "which must be in the range of 800-899",
+                         "SymXchange User Number [800]")
+
+ TARGETCOMMENT=CHARACTERREAD("",
+                             "Enter transaction comment to match. If left",
+                             "blank then 'TC' parameter value",
+                             "from BANNO.M2MTRANSFERS.V3.CFG Letter",
+                             "file will be used",
+                             "Enter transaction comment to target []")
+
+ TARGETSTARTDATE=DATEREAD("",
+                          "Reporting start date [SYSTEMDATE]")
+ IF TARGETSTARTDATE=DATENULL THEN
+  TARGETSTARTDATE=SYSTEMDATE
+
+ TARGETENDDATE=DATEREAD("",
+                        "Reporting end date [SYSTEMDATE]")
+ IF TARGETENDDATE=DATENULL THEN
+  TARGETENDDATE=SYSTEMDATE
+
+ IF TARGETSYMXUSER<800 OR
+    TARGETSYMXUSER>899 THEN
+  DO
+   PRINT FORMAT("Invalid SymXchange user number:999 - TERMINATING PROGRAM RUN.",TARGETSYMXUSER)
+   NEWLINE
+   TERMINATE
+  END
+ ELSE IF TARGETENDDATE<TARGETSTARTDATE OR
+         TARGETSTARTDATE>SYSTEMDATE THEN
+  DO
+   PRINT "Invalid start/end date(s) - TERMINATING PROGRAM RUN."
+   NEWLINE
+   TERMINATE
+  END
+END [PROCEDURE]
+
+PROCEDURE FETCHLETTERFILEPARAMS
+[* Open Letter file BANNO.M2MTRANSFERS.V3.CFG to retrieve needed
+** parameter settings: 
+** 'TC:' (tran comment to search for)
+** 'XA:' (cross accounts allowed flag
+*]
+ XACCTACCESS=FALSE
+
+ FILEOPEN("LETTER",CFGFILENAME,"READ",CFGFILENUMBER,CFGFILEERROR)
+ IF CFGFILEERROR="" THEN
+  DO
+   WHILE (CFGFILEERROR="" AND
+          TARGETCOMMENT="")
```

Comment:
I think the check for XA: needs to be moved out of this while loop sine cross accounts is independent of the target comment, or Just remove the AND TARGETCOMMENT="" from the while loop.